src/lib/controllers/ui-event-controller.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * This is an event controller that is used currently for user interaction event, but can be adapted
+ * to handle events of any type.
+ */
+export default class UIEventController {
+  constructor () {
+    this.listeners = new Map()
+  }
+
+  /**
+   * Registers a single event listener or a group of event listeners.
+   * @param {string} name - A name of the listener or a group of listeners
+   *        that can be used to operate it (register, unregister, set, or remove).
+   * @param {string} nodeOrSelector - A CSS selector to define HTML elements that will listen to an events
+   *        or an HTML Node element.
+   * @param {Function} eventHandler - An event listener function to register.
+   * @param {PointerEvt} EventType - A constructor function of on of a custom pointer events that will do an event handling.
+   * @param {...*} eventExtraParams - Extra arguments that will be passed to the
+   *        PointerEvt constructor function after two obligatory parameters of HTML element and handler function.
+   * @return {UIEventController} An event controller instance for chaining.
+   */
+  registerListener (name, nodeOrSelector, eventHandler, EventType, ...eventExtraParams) {
+    let events
+    if (!nodeOrSelector || nodeOrSelector instanceof Node) {
```

Comment:
So, am I correct that this will need to be refactored when you add the non-DOM events for inter-component communication?

Reply:
My idea is that this will be a specialized controller to handle user interactions only. Inter component communication would be handled by an EventEmitter: https://github.com/alpheios-project/components/blob/ui-controller-refactoring/src/lib/events/event-emitter.js

We could in theory use EventEmitter for handling user interactions, but this will require changing custom pointer events even more and I'm not sure if it will benefit us or not. We probably should postpone a consideration of this for later. What do you think?

Reply:
ok, I think that makes sense. Will be helpful to see the EventEmitter used in practice.